lints/etsi/lint_qcstatem_psd2_roles.go

```diff
@@ -0,0 +1,79 @@
+package etsi
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (ส"License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"encoding/asn1"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/lint"
+	"github.com/zmap/zlint/util"
+)
+
+type qcStatemPsd2Roles struct{}
+
+func (l *qcStatemPsd2Roles) Initialize() error {
+	return nil
+}
+
+func (this *qcStatemPsd2Roles) getStatementOid() *asn1.ObjectIdentifier {
+	return &util.IdEtsiPsd2Statem
+}
+
+func (l *qcStatemPsd2Roles) CheckApplies(c *x509.Certificate) bool {
+	if !util.IsExtInCert(c, util.QcStateOid) {
+		return false
+	}
+	return util.ParseQcStatem(util.GetExtFromCert(c, util.QcStateOid).Value, *l.getStatementOid()).IsPresent()
+
+}
+
+func isValidPSD2Role(oid *asn1.ObjectIdentifier, name *string) bool {
+	return ((oid.Equal(util.IdEtsiPsd2RolePspAs) && *name == "PSP_AS") ||
+		(oid.Equal(util.IdEtsiPsd2RolePspPi) && *name == "PSP_PI") ||
+		(oid.Equal(util.IdEtsiPsd2RolePspAi) && *name == "PSP_AI") ||
+		(oid.Equal(util.IdEtsiPsd2RolePspIc) && *name == "PSP_IC"))
+}
+
+func (l *qcStatemPsd2Roles) Execute(c *x509.Certificate) *lint.LintResult {
```

Comment:
@dadrian / @cpu : Structurally, could you provide guidance on what granularity of lints is expected?

In [TS 119 495](https://www.etsi.org/deliver/etsi_ts/119400_119499/119495/01.04.01_60/ts_119495v010401p.pdf), there are three discrete requirements regarding roles:
* A role must be present (GEN-5.1-2, GEN-5.2.2-1)
* The role must be one of the four OIDs defined (GEN-5.2.2-2)
* The role name must match the abbreviated form for the role OID (GEN-5.2.2-3, REG-5.2.2-5)

This PR currently checks the latter two requirements together, which is not unreasonable, considering there's a 1:1 mapping with OID and name.

However, I wasn't sure whether that was OK, or whether it made sense to write these as three discrete lints, given the three discrete technical requirements. Thoughts?

Reply:
Oh, and I should mention, this is a stand-in since it applies to several other PRs (As somewhat documented in #410 ), which also touch on fields with multiple discrete requirements. This was the smallest one I could find, so I thought it'd be easy to check with this first, as if we split this, then we should also split up any other larger one, presumably.